website_form_recaptcha/tests/test_captcha.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2015-2017 LasLabs Inc.
+# Copyright 2019 Simone Orsi - Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import mock
+
+from odoo.exceptions import ValidationError
+from odoo.tests.common import TransactionCase
+
+imp_model = "odoo.addons.website_form_recaptcha.models.website_form_recaptcha"
+imp_requests = "%s.requests" % imp_model
+
+
+class TestCaptcha(TransactionCase):
+    def setUp(self):
+        super(TestCaptcha, self).setUp()
+        self.model_obj = self.env["website.form.recaptcha"]
+        self.validate_vars = "T1", "T2"
+        self.website = self.env["website"].get_current_website()
+        self.website.recaptcha_key_site = "test-site"
+        self.website.recaptcha_key_secret = "test-secret"
+
+    @mock.patch(imp_requests)
+    def test_post_with_proper_data(self, mk):
+        mk.post.side_effect = StopIteration
+        exp1, exp2 = self.validate_vars
+        try:
+            self.model_obj._validate_response(exp1, exp2, website=self.website)
+        except StopIteration:
+            pass
+        mk.post.assert_called_once_with(
+            self.model_obj.URL,
+            data={
+                "secret": self.website.recaptcha_key_secret,
+                "response": exp1,
+                "remoteip": exp2,
+            },
+        )
+
+    @mock.patch(imp_requests)
+    def test_valid(self, mk):
+        expect = {"success": True}
+        mk.post().json.return_value = expect
+        self.assertTrue(
+            self.model_obj._validate_response(*self.validate_vars, website=self.website)
+        )
+
+    @mock.patch(imp_requests)
+    def test_known_error_raises(self, mk):
+        expect = {"error-codes": ["missing-input-secret"]}
+        mk.post().json.return_value = expect
+        with self.assertRaises(ValidationError):
+            self.model_obj._validate_response(*self.validate_vars, website=self.website)
+
+    @mock.patch(imp_requests)
+    def test_known_error_lookup(self, mk):
+        expect = {"error-codes": ["missing-input-secret"]}
+        mk.post().json.return_value = expect
+        try:
+            self.model_obj._validate_response(*self.validate_vars, website=self.website)
+        except ValidationError as e:
+            self.assertEqual(
+                e.name, self.model_obj._get_error_message(expect["error-codes"][0])
```

Comment:
@AreebSiddiqiIR 

 - The warning [2021-05-16 22:26:32,126 5813 WARNING openerp_test py.warnings: /home/travis/odoo-14.0/odoo/exceptions.py:38: DeprecationWarning: UserError attribute 'name' is a deprecated alias to args](https://travis-ci.com/github/OCA/website/jobs/505747440#L1033) is because of this line:

 - `e.name`

It means the exception has deprecated that old attribute for odoo v14.0 or python3

Could you check what is the new attribute to use here, please?